packages/hooks/src/useMutationObserver.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react';
+
+type MutationHandler<Value> = (
+  mutations: Array<MutationRecord>,
+  observer: MutationObserver,
+) => Value;
+
+/**
+ * Hook to subscribe to changes on the DOM.
+ * @param target HTML element that is subscribed to DOM changes.
+ * @param options Object with information about what DOM changes to subscribe to.
+ * @param callback Callback function to execute inside of MutationObserver instance.
+ * @param enabled Determines whether or not the hook should run, defaults to true.
+ */
+export default function useMutationObserver<Value>(
+  target: HTMLElement | null,
+  options: MutationObserverInit,
```

Comment:
Options should probably be optional, and come after `callback`.

Reply:
*Unless it's required for initialization

Reply:
based on the docs it does look like options is optional, but @hswolff wrote this one so want to make sure he agrees

Reply:
Based on the docs it's optional but when you run it in the browser you get an error without the object present.

I think I also did it this way to force the user of this function to think about what they wanted to observe, so you didn't get the default behavior which may surprise a user.

![image](https://user-images.githubusercontent.com/322576/58973929-29086580-878f-11e9-81a6-a930f1886803.png)

Reply:
Okay, if it errors without an object present, I buy this! Thanks for the note Harry 👍